auth0/resource_auth0_action.go

```diff
@@ -251,6 +259,61 @@ func deployAction(ctx context.Context, d *schema.ResourceData, m interface{}) di
 	return diag.FromErr(d.Set("version_id", actionVersion.GetID()))
 }
 
+func preventErasingUnmanagedSecrets(d *schema.ResourceData, api *management.Management) diag.Diagnostics {
+	if !d.HasChange("secrets") {
+		return nil
+	}
+
+	preUpdateAction, err := api.Action.Read(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	oldSecrets, newSecrets := d.GetChange("secrets")
+
+	return checkForUnmanagedActionSecrets(
+		oldSecrets.([]interface{}),
+		newSecrets.([]interface{}),
+		preUpdateAction.Secrets,
+	)
+}
+
+func checkForUnmanagedActionSecrets(
+	oldSecretsFromConfig []interface{},
+	newSecretsFromConfig []interface{},
+	secretsFromAPI []*management.ActionSecret,
+) diag.Diagnostics {
+	secretKeysInConfigMap := make(map[string]bool, len(secretsFromAPI))
+
+	for _, oldSecret := range oldSecretsFromConfig {
+		secretKeyName := oldSecret.(map[string]interface{})["name"].(string)
+		secretKeysInConfigMap[secretKeyName] = true
+	}
+
+	for _, newSecret := range newSecretsFromConfig {
+		secretKeyName := newSecret.(map[string]interface{})["name"].(string)
+		secretKeysInConfigMap[secretKeyName] = true
+	}
+
+	var warnings diag.Diagnostics
+	for _, secret := range secretsFromAPI {
+		if _, ok := secretKeysInConfigMap[secret.GetName()]; !ok {
+			warnings = append(warnings, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Unmanaged Action Secret",
+				Detail: fmt.Sprintf(
+					"Found unmanaged action secret with key: %s. "+
+						"Add this secret to your configuration so it does not get wiped.",
```

Comment:
```suggestion
					"Detected an action secret not managed though Terraform: %s. If you proceed, this secret will get deleted. It is required to add this secret to your action configuration to prevent unintentionally destructive results.",
```